src/app/components/post-detail/post-detail.component.ts

```diff
@@ -17,30 +18,51 @@ export class PostDetailComponent implements OnInit {
 
   private isMyPost: boolean = false;
 
-  constructor(private postService: PostService, private route: ActivatedRoute, private authService: AuthService) { }
+  constructor(private postService: PostService,
+              private route: ActivatedRoute, private authService: AuthService,
+              private loc: Location) { }
 
   ngOnInit() {
       this.route.params.subscribe(params => {
       this.postService.getPost(params['_id'])
-        .then(post => this.post = post);
+        .then(post => this.initPostStatus(post));
+
     });
+
+
+
+  }
+  initPostStatus(post: Post){
+    this.post = post;
```

Comment:
Why pass the post to this function, instead of just keeping the `this.post = post` assignment in `ngOnInit()`?

Reply:
because in order to check whether this post is my post, i must wait for the actual post to be returned from backend. i found checking outside of the scope of "then" did not work because post was null (async operations?)

Reply:
cade?